src/org/exist/xquery/XQueryContext.java

```diff
@@ -1418,8 +1418,6 @@ public void reset(final boolean keepGlobals) {
 
         clearUpdateListeners();
 
-        cleanupTasks.clear();
```

Comment:
think that call `runCleanupTasks` instead with comment will be good? As minimum it'll avoid regression as it happen [here](https://github.com/ChiSamurai/exist/commit/0301cdfc768e5014bdf1132dce93cf467b58638a).